### ISSUE TITLE
nextcloud deployement fails with caching.apcu  set to false

### ISSUE THREAD

**Issue by teto** [NixOS/nixpkgs#242675]
### Describe the bug

when I try to deploy my server with deploy-rs:
```
Jul 10 17:26:29 neotokyo systemd[1]: Starting nextcloud-add-user.service...
Jul 10 17:26:30 neotokyo nextcloud-setup-start[33153]: An unhandled exception has been thrown:
Jul 10 17:26:30 neotokyo nextcloud-setup-start[33153]: OCP\HintException: [0]: Memcache \OC\Memcache\APCu not available for local cache (Is the matching PHP module installed and enabled?)
```

### Steps To Reproduce

```
  services.nextcloud = {
    enable = true;
    package = pkgs.nextcloud27;

    # Use HTTPS for links
    https = true;

    # New option since NixOS 23.05
    configureRedis = true;
    caching.apcu = false;  ### <<< When set to false I get the previous error, with true (the default), it works fine.

    config = {
      # Further forces Nextcloud to use HTTPS
      overwriteProtocol = "https";
      # loaded via sops 
      adminpassFile = "/run/secrets/${nextcloudAdminPasswordSopsPath}";
    };
    maxUploadSize = "512M";
    logLevel = 0;
    enableBrokenCiphersForSSE = false;
    # increase security
    enableImagemagick = false;
    autoUpdateApps.enable = true;

    extraApps = with config.services.nextcloud.package.packages.apps; {


     # inherit news; # removed 'cos gives a wrong error
     inherit contacts;
    };
    extraAppsEnable = true;
  };

  networking.firewall.allowedTCPPorts = [ 80 443 ];

```

### Expected behavior
deploy ok

### Additional context
what it is a bit troubling to me is that the same config 

### Notify maintainers

@Ma27 @Misterio77 

### Metadata

nixos-unstable  408c0e8c15a1c9cf5c3226931b6f283c9867c484

NB: unrelated but logs contain 
```
Jul 10 23:29:53 neotokyo Nextcloud[36584]: {"reqId":"wfn0gXaD7BYqFt7XDmS9","level":0,"time":"2023-07-10T21:29:53+00:00","remoteAddr":"","user":"--","app":"serverDI","method":"","url":"--","message":"The requested alias \"SystemConfig\" is deprecated. Please request \"OC\\SystemConfig\" directly. This alias will be removed in a future Nextcloud version.","userAgent":"--","version":"27.0.0.8","data":{"app":"serverDI"}}
```
I am not sure where that comes from

**Comment by Ma27** [comment]
`configureRedis` only sets `memcache.distributed` & `memcache.locking` to `\OC\Memcache\Redis` while `memcache.local` remains using APCu.

See also the administrator manual: https://docs.nextcloud.com/server/27/admin_manual/configuration_server/caching_configuration.html

Basically, it's recommended to use Redis in case of larger setups (or clustered setups), but even then the upstream recommendation is to use APCu for local caching:

> APCu is faster at local caching than Redis. If you have enough memory, use APCu for Memory Caching and Redis for File Locking. If you are low on memory, use Redis for both.

So basically, `configureRedis` follows the upstream recommendation. Of course, you're free to use Redis for local caching as well, but you have to configure it manually:

```
services.nextcloud.extraOptions."memcache.local" = ''\OC\Memcache\Redis'';
```
